customization-base/src/main/java/com/azure/autorest/customization/ConstantCustomization.java

```diff
@@ -0,0 +1,183 @@
+        SymbolInformation symbol, String constantName) {
+        super(editor, languageClient, symbol);
+
+        this.packageName = packageName;
+        this.className = className;
+        this.constantName = constantName;
+    }
+
+    /**
+     * Gets the name of the class that contains this constant.
+     *
+     * @return The name of the class that contains this constant.
+     */
+    public String getClassName() {
+        return className;
+    }
+
+    /**
+     * Gets the name of this constant.
+     *
+     * @return The name of this constant.
+     */
+    public String getConstantName() {
+        return constantName;
+    }
+
+    /**
+     * Gets the Javadoc customization for this constant.
+     *
+     * @return The Javadoc customization.
+     */
+    public JavadocCustomization getJavadoc() {
+        return new JavadocCustomization(editor, languageClient, fileUri, fileName,
+            symbol.getLocation().getRange().getStart().getLine());
+    }
+
+    /**
+     * Replace the modifier for this constant.
+     * <p>
+     * For compound modifiers such as {@code public abstract} use bitwise OR ({@code |}) of multiple Modifiers, {@code
+     * Modifier.PUBLIC | Modifier.ABSTRACT}.
+     * <p>
+     * This operation doesn't allow for the constant to lose constant status, so
+     * {@code Modifier.STATIC | Modifier.FINAL} will be added to the passed {@code modifiers}.
+     * <p>
+     * Pass {@code 0} for {@code modifiers} to indicate that the constant has no modifiers.
+     *
+     * @param modifiers The {@link Modifier Modifiers} for the constant.
+     * @return The updated ConstantCustomization object.
+     * @throws IllegalArgumentException If the {@code modifier} is less than to {@code 0} or any {@link Modifier}
+     * included in the bitwise OR isn't a valid constant {@link Modifier}.
+     */
+    public ConstantCustomization setModifier(int modifiers) {
+        replaceModifier(symbol, editor, languageClient, "(?:.+ )?(\\w+ )" + constantName + "\\(",
+            "$1" + constantName + "(", Modifier.fieldModifiers(), Modifier.STATIC | Modifier.FINAL | modifiers);
+
+        return refreshCustomization(constantName);
+    }
+
+    /**
+     * Renames the constant.
+     * <p>
+     * This operation doesn't allow for the constant to lose naming conventions of capitalized and underscore delimited
+     * words, so the {@code newName} will be capitalized.
+     * <p>
+     * This is a refactor operation, all references of the constant will be renamed and the getter method(s) for this
+     * property will be renamed accordingly as well.
+     *
+     * @param newName The new name for the constant.
+     * @return A new instance of {@link ConstantCustomization} for chaining.
+     * @throws NullPointerException If {@code newName} is null.
+     */
+    public ConstantCustomization rename(String newName) {
```

Comment:
I don't see an API to change the value of the constant. Should that be included too?

Reply:
Also can be added in the future